wcomponents-theme/src/main/js/wc/file/util.js

```diff
@@ -0,0 +1,101 @@
+define(["wc/dom/uid", "wc/file/getMimeType"], function(uid, getMimeType) {
+	
+	var instance = new FileUtil();
+	/**
+	 * Used when checking the newly created image is named with the correct extension.
+	 * If it does not already match any in the array then the extension at index zero will be appended.
+	 **/
+	FileUtil.prototype.mimeToExt = {
+		"image/jpeg": ["jpeg", "jpg"],
+		"image/png": ["png"],
+		"image/webp": ["webp"]
+	};
+
+	function FileUtil() {
+		/**
+		 * Converts a generic binary blob to a File blob.
+		 * @param {Blob} blob
+		 * @param {Object} [config] Attempt to set some of the file properties such as "type", "name"
+		 * @returns {File} The File blob.
+		 */
+		this.blobToFile = function (blob, config) {
+			var name,
+				filePropertyBag = {
+					type: blob.type,
+					lastModified: new Date()
+				};
+			if (config) {
+				if (!filePropertyBag.type) {
+					filePropertyBag.type = config.type;
+				}
+				name = config.name;
+			}
+			name = name || uid();
+
+
+//			if (typeof File === "function") {
+//				return new File([blob], name, filePropertyBag);
+//			}
```

Comment:
Clean up this - remove the stuff you don't use and reduce the empty space. This is new: there is no need to copy over the previous thought processes.